File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        prod = 1
        zeroes = 0

        for num in nums:
            if num == 0:
                zeroes += 1
                if zeroes > 1:
                    return [0]*len(nums)
            else:
                prod *= num


        for i in range(len(nums)):
            if zeroes == 1:
                if nums[i] != 0:
                    nums[i] = 0
                else:
                    nums[i] = prod
            else:
                nums[i] = prod//nums[i]
        
        return nums

        

File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {

        int length = nums.size()*2;
        vector<int> arr;
        
        for (int i = 0; i < length; i++){
            if (i < nums.size()){
                arr.push_back(nums[i]);
            }
            else{
                arr.push_back(nums[i - nums.size()]);
            }
        }

        return arr;
    }

};